sonar-python-plugin/src/main/java/org/sonar/plugins/python/pylint/PylintSensor.java

```diff
@@ -103,18 +103,17 @@ protected void analyzeFile(File file, File out) throws IOException {
     List<Issue> issues = analyzer.analyze(file.getAbsolutePath(), fileSystem.encoding(), out);
 
     for (Issue pylintIssue : issues) {
-      ActiveRule rule = activeRules.find(RuleKey.of(PylintRuleRepository.REPOSITORY_KEY, pylintIssue.getRuleId()));
+      ActiveRule rule = context.activeRules().find(RuleKey.of(PylintRuleRepository.REPOSITORY_KEY, pylintIssue.getRuleId()));
 
       if (rule != null) {
-        Issuable issuable = resourcePerspectives.as(Issuable.class, pyFile);
-        if (issuable != null) {
-          org.sonar.api.issue.Issue issue = issuable.newIssueBuilder()
-              .ruleKey(rule.ruleKey())
-              .line(pylintIssue.getLine())
-              .message(pylintIssue.getDescription())
-              .build();
-          issuable.addIssue(issue);
-        }
+        NewIssue newIssue = context
+          .newIssue()
+          .forRule(rule.ruleKey());
+        newIssue.at(newIssue.newLocation()
+          .on(pyFile)
+          .at(pyFile.selectLine(pylintIssue.getLine()))
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: A "NullPointerException" could be thrown; "pyFile" is nullable here. ([squid:S2259](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.python%3Apython&branch=SONARPY-235b&issues=AWDqhCANo_TTbi26M1GJ&open=AWDqhCANo_TTbi26M1GJ)

[ISSUE_KEY]: AWDqhCANo_TTbi26M1GJ